pallets/vector/src/lib.rs

```diff
@@ -401,72 +414,79 @@ pub mod pallet {
 	where
 		[u8; 32]: From<T::AccountId>,
 	{
-		/// The entrypoint for fulfilling a call.
-		/// function_id Function identifier.
-		/// input Function input.
-		/// output Function output.
-		/// proof  Function proof.
-		/// slot  Function slot to update.
 		#[pallet::call_index(0)]
+		// Unused parameter that's hard to remove, would mess up traits
 		#[pallet::weight(weight_helper::fulfill_call::<T>(* function_id))]
 		pub fn fulfill_call(
 			origin: OriginFor<T>,
 			function_id: H256,
-			input: FunctionInput,
-			output: FunctionOutput,
-			proof: FunctionProof,
+			proof_bytes: Vec<u8>,
+			public_values: Vec<u8>,
 			#[pallet::compact] slot: u64,
 		) -> DispatchResultWithPostInfo {
-			let sender: [u8; 32] = ensure_signed(origin)?.into();
-			let updater = Updater::<T>::get();
-			// ensure sender is preconfigured
-			ensure!(H256(sender) == updater, Error::<T>::UpdaterMisMatch);
+			
+			let proof_outputs: ProofOutputs = SolValue::abi_decode(&public_values, true).unwrap();
+			let head = Head::<T>::get();
+			let new_head: [u8; 8] = proof_outputs.newHead.to_le_bytes(); 
+			let new_head = u64::from_le_bytes(new_head[..8].try_into().unwrap());
+			ensure!(new_head <= head, Error::<T>::SlotBehindHead);
 
-			let config = ConfigurationStorage::<T>::get();
-			let input_hash = H256(sha2_256(input.as_slice()));
-			let output_hash = H256(sha2_256(output.as_slice()));
-			let (step_function_id, rotate_function_id) = Self::get_function_ids()?;
-			let verifier = Self::get_verifier(function_id, step_function_id, rotate_function_id)?;
+			let vkey_hash = "0x00788ce8dc2970920a3d3c072c8c07843d15f1307a53b3dd31b113c3e71c28e8";
```

Comment:
Make this a constant outside of the function and rename it to `SP1_HELIOS_VKEY`.